### ISSUE TITLE
Notes on compiling preForth on Fedora 40 (x86_64)

### ISSUE THREAD

**Issue by cstrotm** [uho/preForth#17]
* Compiling preForth and SeedForth on Fedora 40 (x86_64)

 * Installing devtools
   
   sudo dnf install gcc make git

 * Installing "fasm" binary from homepage
   
 * Installing i686 libraries

   sudo dnf install glibc-devel.i686 glibc-devel

 * Adjust Makefile

```
   # assemble and link executable on linux
   %.Linux: %.asm
        fasm $< $@.o
        LDEMULATION=elf_i386 ld -melf_i386 -o $@ \
           -dynamic-linker /lib/ld-linux.so.2 \
           /usr/lib/crt1.o /usr/lib/crti.o \
           $@.o \
           -lc /usr/lib/crtn.o
```
   

 * Build preForth and seedForth

   make clean
   make
   ./seed

**Comment by znmeb** [comment]
I have Fedora 40 (Silverblue - can run any Linux in a container) - does this work there? I'm having trouble getting it to work with Ubuntu 24.04 LTS.

**Comment by cstrotm** [comment]
Fedora Silverblue is an immutable OS image system. You would need to use a toolbox container to install and run PreForth/SeedForth.

I can check with Ubuntu 24.04 LTS later today and report if I can get PreForth working there (should work)

**Comment by znmeb** [comment]
Yes, I know all about the immutable thing - that's why I picked it! I'm actually on Universal Blue Bluefix DX, which is a superset of Sliverblue. I have a Fedora 40 distrobox, a Rawhide distrobox, about three Ubuntu 24.04 distroboxes and an Arch distrobox at the moment. Plus Ubuntu 24.04 in Windows Subsystem for Linux. I'll try this tomorrow on Fedora and Arch.

**Comment by cstrotm** [comment]
I've added a new issue #18 on how to compile perForth and seedForth on Ubuntu 14.04 LTS.
